features_api_database/runtime/handler.py

```diff
@@ -72,38 +72,79 @@ def get_secret(secret_name):
-    else:
-        print(f"database {db_name} not found, creating...")
-        cursor.execute(
-            sql.SQL("CREATE DATABASE {db_name}").format(db_name=sql.Identifier(db_name))
-        )
+    print(f"DEBUG: create_db called with db_name='{db_name}'")
+
+    try:
+        cursor.execute("SELECT datname FROM pg_catalog.pg_database ORDER BY datname")
+        existing_dbs = [row[0] for row in cursor.fetchall()]
+
+        print(f"DEBUG: Existing databases: {existing_dbs}")
+
+        if db_name in existing_dbs:
+            print(f"DEBUG: Database '{db_name}' already exists")
+        else:
+            print(f"DEBUG: Database '{db_name}' not found, creating...")
+            cursor.execute(
+                sql.SQL("CREATE DATABASE {}").format(sql.Identifier(db_name))
+            )
+            print(f"DEBUG: Database '{db_name}' created successfully")
+
+            cursor.execute("SELECT datname FROM pg_catalog.pg_database WHERE datname = %s", [db_name])
+            if cursor.fetchone():
+                print(f"DEBUG: Database '{db_name}' exists after creation")
+            else:
+                print(f"DEBUG: Database '{db_name}' was NOT created successfully")
+
+    except Exception as e:
+        print(f"ERROR: create_db failed: {e}")
+        raise
 
 
 def create_user(cursor, username: str, password: str) -> None:
     """Create User."""
-    cursor.execute(
-        sql.SQL(
-            "DO $$ "
-            "BEGIN "
-            "  IF NOT EXISTS ( "
-            "       SELECT 1 FROM pg_roles "
-            "       WHERE rolname = {user}) "
-            "  THEN "
-            "    CREATE USER {username} "
-            "    WITH PASSWORD {password}; "
-            "  ELSE "
-            "    ALTER USER {username} "
-            "    WITH PASSWORD {password}; "
-            "  END IF; "
-            "END "
-            "$$; "
-        ).format(username=sql.Identifier(username), password=password, user=username)
-    )
+
+    print(f"DEBUG: create_user called with username='{username}'")
+
+    try:
+        # Check if user exists before
+        cursor.execute("SELECT rolname FROM pg_roles WHERE rolname = %s", (username,))
+        exists_before = cursor.fetchone() is not None
+        print(f"DEBUG: User '{username}' exists before: {exists_before}")
+
+        # Create/update user
+        cursor.execute(
+            sql.SQL(
+                "DO $$ "
+                "BEGIN "
+                "  IF NOT EXISTS ( "
+                "       SELECT 1 FROM pg_roles "
+                "       WHERE rolname = {user}) "
+                "  THEN "
+                "    CREATE USER {username} "
+                "    WITH PASSWORD {password}; "
+                "  ELSE "
+                "    ALTER USER {username} "
+                "    WITH PASSWORD {password}; "
+                "  END IF; "
+                "END "
+                "$$; "
+            ).format(username=sql.Identifier(username), password=password, user=username)
```

Comment:
https://realpython.com/prevent-python-sql-injection/#exploiting-query-parameters-with-python-sql-injection

```suggestion
            ).format(username=sql.Identifier(username), password=sql.Identifier(password), user=sql.Identifier(username))
```

also I think the debug executed below needs to be wrapped in `sql.SQL` and updated to use `sql.Identifier`